packages/xstate-angular/src/injectActor.ts

```diff
@@ -0,0 +1,57 @@
+import { createActor } from 'xstate';
+import type {
+  ActorOptions,
+  ActorRefFrom,
+  AnyActorLogic,
+  SnapshotFrom
+} from 'xstate';
+import {
+  DestroyRef,
+  Injector,
+  inject,
+  runInInjectionContext,
+  signal,
+  Signal
+} from '@angular/core';
+
+interface injectActorOptions {
+  injector?: Injector;
+}
+
+interface InjectedActor<TLogic extends AnyActorLogic> {
+  snapshot: Signal<SnapshotFrom<TLogic>>;
+  send: ActorRefFrom<TLogic>['send'];
+  ref: ActorRefFrom<TLogic>;
+}
+
+export function injectActor<TLogic extends AnyActorLogic>(
+  logic: TLogic,
+  options?: ActorOptions<TLogic>,
+  injectOptions?: injectActorOptions
+): InjectedActor<TLogic> {
+  const injector = injectOptions?.injector ?? inject(Injector);
+  const destroyRef = injector.get(DestroyRef);
+  // I'm afraid I stepped into this: https://github.com/angular/angular/issues/34478
+  return runInInjectionContext(injector, () => {
+    const actorInstance = createActor(logic as any, options).start();
+
+    const snapshot = signal(actorInstance.getSnapshot());
+    const result = {
+      snapshot,
+      send: actorInstance.send,
+      ref: actorInstance
+    } as any;
+
+    const subscription = actorInstance.subscribe((currentSnapshot) => {
+      // result.snapshot = snapshot;
+      snapshot.set(currentSnapshot);
+    });
+
+    destroyRef.onDestroy(() => {
+      actorInstance.stop();
+      subscription.unsubscribe();
```

Comment:
I usually find it much less error-prone to dispose of local things first before stopping the external~ resource. It usually leads to removing some edge cases around synchronous notification from within the stopping call or ones related to the fact that the other call might throw and interrupt the whole cleanup. I don't think we have to wrap this in any try/catches or nothing - nothing here should be able to throw (if it does then I'd consider that to be a likely bug in XState). It's just a rule of thumb that I developed over the years and try to follow as much as possible
```suggestion
      subscription.unsubscribe();
      actorInstance.stop();
```